### DISCUSSION TITLE
Error during Vaultwarden compilation

### DISCUSSION THREAD

**Discussion by Nyxtorm** [dani-garcia/vaultwarden#2056] [Q&A]
### Subject of the issue
Error during Vaultwarden compilation

### Deployment environment

* vaultwarden version: 1.22.2-b4c95fb4

* Install method: built from source

* Clients used: web vault, desktop, Android

* Reverse proxy and version: nginx 1.21.0

* MySQL/MariaDB or PostgreSQL version: SQLite: 3.35.4

* Other relevant details: OS Debian 10 w/ backports

### Steps to reproduce

* Clone all necessary repositories
* Update Rust with rustup to the latest nightly update
* Start the Vaultwarden build:
```
cargo build --features sqlite --release
```

### Expected behaviour
Successful compilation

### Actual behaviour
Error after crates.io index update. I have tested with the stable, beta and nightly versions of rustup.

### Troubleshooting data

```
$ cargo build --features sqlite --release
info: syncing channel updates for 'nightly-2021-10-14-x86_64-unknown-linux-gnu'
info: latest update on 2021-10-14, rust version 1.57.0-nightly (dfc5add91 2021-10-13)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'
    Updating crates.io index
error: failed to select a version for the requirement `reqwest = "^0.11.5"`
candidate versions found which didn't match: 0.11.4, 0.11.3, 0.11.2, ...
location searched: crates.io index
required by package `vaultwarden v1.0.0 (/volumes/data/vaultwarden/vw-compile/vaultwarden)`
```

**Comment by BlackDex** [comment]
This is not something we can fix. This is either something with your internet connection, or something wrong on the side of crates.io or maybe even cargo.

I just removed and cleared my whole cargo cache and it build without issues.
Also, yesterday there was released a new version which also compiled without issues.

I suggest to do a `cargo clean` and try it again to see what happens.

**Reply by BernhardFahr** [reply]
I tried to follow the recommendation (cargo clean). Has unfortunately brought nothing.
The following worked for me:
1. delete .cargo (in the home directory).
2. deleting .rustc (in the home directory).
3. reinstallation (curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh)

After that the compilation went through without errors.

**Reply by Nyxtorm** [reply]
@BlackDex @BernhardFahr Thank you for your answers which allowed me to correct my problem.. Indeed, the `cargo clean` was not enough.

I have removed `~/.cargo` and `~/.rustup` and reinstalled a clean Rustup. The build now sees all new versions of reqwest.

Thanks for your help! 👍

**Reply by BlackDex** [reply]
You can also try to install cargo-cache, which has an option to remove this. `cargo install cargo-cache` and after that use `cargo cache -r all`

**Reply by Nyxtorm** [reply]
I don't know cargo very well yet, thanks for the tip, I just installed it to test next times.

**Reply by asanDD** [reply]
Reinstalling rustup / cargo solved the problem for me too. Thank you all.
